src/compiler.ml

```diff
@@ -105,27 +105,37 @@ type table = {
   uuid : Util.UUID.t;
 } [@@deriving show]
 
-type pruned_table = {
-  c2s0 : string D.Constants.Map.t;
-  c2t0 : D.term D.Constants.Map.t;
-} [@@deriving show]
+type entry =
+| ENeg of D.constant * string
+| EPos of D.constant * D.term
```

Comment:
ENEg should be something like GlobalSymbol (* < 0 *)
EPos should be BoundVariable (* >= 0 *)

Reply:
I'm a bit surprise it is not 2 arrays, IIRC the bound variable one is small and the other is contiguous, so using the abs of the value as the array position seems more natural/understandable and would remove half of the data in the buckets

Reply:
I mean, D.constant is an int, -1 ... -maxg for globals, 0 .. maxl for locals, where maxl is typically not more than 3, while maxg can be large (thousands)

Reply:
I went for the most straightforward changes, we can try to go even further. I don't think that splitting the array will matter much though.

> the other is contiguous

Is it still the case after pruning? The set of live symbols could also contain globals IIUC.

Reply:
(I did change the constructor names for clarity.)